lstchain/scripts/lstchain_longterm_dl1_check.py

```diff
@@ -1507,8 +1507,31 @@ def pixel_report(title, value, low_limit, upp_limit, run_fraction):
     return
 
 def get_datacheck_table(filename, tablename, exclude_stars=True):
+    """
+
+    Parameters
+    ----------
+    filename: str, datacheck_dl1_LST-1.RunXXXXX.h5 full-run datacheck file
+    tablename: str, "pedestals" "cosmics" or "flatfield"
+    exclude_stars: set to nan all pixel values for subruns (table rows) in
+    which the given pixel had stars nearby, according to colum num_nearby_stars
+
+    Returns
+    -------
+
+
+    """
     table = read_table(filename, f'/dl1datacheck/{tablename}')
 
+    # Set to nan pixel entries for which there were nearby stars:
+    for k in table.keys():
+        if k == 'num_nearby_stars':
+            continue
+        if table[k].shape != table['num_nearby_stars'].shape:
+            continue
+        table[k] = np.where(table['num_nearby_stars'] > 0,
```

Comment:
This overwrites the full column with a new copy instead of just setting the values to nan.

Also: astropy supports masked columns, which might be better then using `nan`, since that also works for integers and strings, where nan does not.

```
In [11]: from astropy.table import Table, MaskedColumn

In [12]: import numpy as np

In [13]: t = Table({'a': [1, 2, 3, 4], 'stars': [1, 0, 5, 0]})

In [14]: mask = t['stars'] > 0

In [15]: t['a'] = MaskedColumn(t['a'], mask=mask, copy=False)

In [16]: t
Out[16]: 
<Table length=4>
  a   stars
int64 int64
----- -----
   --     1
    2     0
   --     5
    4     0

In [17]: t['a'].mean()
Out[17]: 3.0
```

Reply:
Note that mask here as the opposite meaning of `array[mask]`. Values with `True` are masked. Like with `np.ma.masked_array`

Reply:
This however changes the behavior - when no event survives the stars cut, I had 'nan' before (so I could later easily ignore those in higher-level averages, using np.nanmean or whatever), now I get 'masked' and this somehow is interpreted as 0's later.

I could simply do `t['a'][t['num_nearby_stars']>0] = np.nan`, right?

Reply:
Ah, but I still have the problem with the integers...

Reply:
> This however changes the behavior - when no event survives the stars cut, I had 'nan' before (so I could later easily ignore those in higher-level averages, using np.nanmean or whatever), now I get 'masked' and this somehow is interpreted as 0's later.

That's strange. You can explicitly convert masked to nan using `masked_array.filled(np.nan)` this will give a normal numpy array with masked values replaced by nan. For positive intergers you could do `masked_array.filled(-1)`

Reply:
```

In [8]: t['a'].filled(-1)
Out[8]: 
<Column name='a' dtype='int64' length=4>
-1
 2
-1
 4
```

Reply:
```
In [9]: t['a'] = MaskedColumn(t['a'], mask=np.ones(4, dtype=bool), copy=False)

In [10]: t
Out[10]: 
<Table length=4>
  a   stars
int64 int64
----- -----
   --     1
   --     0
   --     5
   --     0

In [11]: t['a'].mean()
Out[11]: masked

In [12]: t['a'].mean().filled(np.nan)
Out[12]: array(nan)
```

Reply:
I went back to the previous code, because I see no drawback in converting integer columns to floats, and this solves the issue that masked arrays are not properly shown in the camera displays (masked entries appear as zeros, which obviously is misleading)

Reply:
I had not seen your replies when I posted my previous comments. Still, I see no drawback in the way I am doing it now, and setting -1 would still be misleading when I camera-display an integer quantity.

Reply:
> masked arrays are not properly shown in the camera displays (masked entries appear as zeros, which obviously is misleading)

Masked values should be shown using the "bad" color of the color map. 

```
cmap = plt.get_cmap("inferno").with_extremes(bad="grey")
```

should show them in gray.